gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java

```
@@ -447,11 +455,11 @@ public void launchJob(@Nullable JobListener jobListener) throws JobException {
       if (this.runningMap.replace(this.jobContext.getJobName(), false, true)) {
         LOGGER.info("Job {} will be executed, add into running map.", this.jobContext.getJobId());
         isLaunched = true;
-        super.launchJob(jobListener);
+        launchJobImpl(jobListener);
       } else {
         LOGGER.warn("Job {} will not be executed because other jobs are still running.", this.jobContext.getJobId());
       }
-      // TODO: Better error handling
+      // TODO: Better error handling. The current impl swallows exceptions for jobs that were started by this method call
     } catch (Throwable t) {
```

Add todo here to make the behavior of swallowing exception configurable